File: app/src/main/java/com/gitsearch/data/model/PaginationLink.java
package com.gitsearch.data.model;

import android.net.Uri;
import android.text.TextUtils;

import com.gitsearch.data.remote.GitHub;

/**
 * User: Christian Krueger
 * Date: 20.01.2018
 *
 * @since 0.5.2
 */
public class PaginationLink {

  private final String rel;
  private final Integer page;

  public PaginationLink(String link) {
    link = link.trim()
      .replace("<", "")
      .replace(">", "")
      .replace("\"", "");
    String url = link.split(";")[0];

    Uri uri = Uri.parse(url);
    String pageParameter = uri.getQueryParameter(GitHub.PARAMETER_PAGE);

    if (TextUtils.isEmpty(pageParameter)) {
      page = null;
    }
    else {
      page = Integer.valueOf(pageParameter);
    }

    rel = link.split("rel=")[1];
  }

  public String getRel() {
    return rel;
  }

  public int getPage() {
    return page;
  }
}

File: app/src/main/java/com/gitsearch/data/model/PaginationLink.kt
package com.gitsearch.data.model

import android.net.Uri
import android.text.TextUtils
import com.gitsearch.data.remote.GitHub

const val PARAMETER_PAGE = "page"

data class PaginationLink(private val link: String) {

    val rel: String
    val page: Int

    init {
        val value = link.trim { it <= ' ' }
                .replace("<", "")
                .replace(">", "")
                .replace("\"", "")

        val url = value.split(";".toRegex())[0]
        val uri = Uri.parse(url)
        val pageParameter = uri.getQueryParameter(PARAMETER_PAGE)

        page = if (pageParameter.isNullOrBlank()) 0 else pageParameter.toInt()
        rel = value.split("rel=".toRegex())[1]
    }

}
